### PR TITLE
Add problems

### PR DESCRIPTION
**Pull request by madhusdhnn**

### CODE REVIEW

File: src/main/java/org/example/problems/PizzaEating.java
Comment (venkatyuva) [inline src/main/java/org/example/problems/PizzaEating.java:1]: Screenshot 2023-07-20 at 12 36 05 PM
A : [ 9330, 2294, 3720, 3472, 8821, 5765, 1080, 2631, 4411, 8081, 6114, 3343, 8139, 8383, 6045, 8829, 8394, 3576, 981, 9001, 5597, 5948, 7371, 6088, 6544, 2734, 2853, 262, 6825, 4401, 2381, 779, 1858, 3424, 9499, 7015, 2846, 1663, 8510, 6611, 6850, 9219, 9291, 5056, 5744, 8556, 6386, 6535, 6276, 7322, 1152, 1227, 5252, 4460, 2784, 6715, 6438, 7880, 7699, 4101, 3138, 4286, 5879, 8285, 343, 4916, 5961, 3068, 5141, 708, 3567, 729, 563, 6364, 1970, 2891, 2624, 2279, 7314, 5762, 7011, 4841, 6979, 6001, 7395, 4508, 7657, 6552, 1787, 3432, 4545, 740, 8777, 7585, 4253, 4109, 5515, 1455, 5020, 3029, 6672, 2371, 6664, 7948, 4025, 4544, 6855, 6009, 4982, 553, 6289, 8164, 8360, 1020, 5380, 5357, 9462, 8488, 8354, 3900, 5789, 4245, 9881, 2467, 8284, 5275, 237, 8910, 8568, 1835, 7940, 9685, 9652, 9550, 9267, 7468, 111, 9590, 6328, 7774, 5582, 9296, 7937, 7875, 5436, 8232, 6973, 9799, 9071, 4588, 6298, 9090, 2647, 9091, 8112, 6613, 7464, 2113, 3782, 1952, 9231, 3322, 3214, 5129, 1951, 8601, 8960, 9755, 9993, 5195, 3331, 8633, 3560, 8778, 7780, 3273, 6384, 9082, 7154, 2623, 8522, 7957, 9819, 5592, 6901, 197, 4121, 2705, 3600, 7130, 7145, 9176, 871, 8417, 4600, 7081, 4619, 4289, 9453, 8129, 2801, 7663, 9150, 6584, 4516, 660, 4879, 5960, 8562, 8049, 4055, 9441, 1382, 8771, 2029, 252, 7545, 5870, 7596, 2470, 2711, 8173, 5924, 2126, 6220, 8851, 1432, 1244, 7847, 6671, 1395, 4611, 6393, 1065, 9525, 9362, 9520, 3621, 1563, 1662, 3726, 9435, 5950, 2091, 7323, 6261, 94, 9929, 5123, 6382, 6367, 6469, 1024, 2723, 6640, 3518, 8794, 9365, 9179, 1957, 6772, 8342, 1356, 837, 8490, 9729, 7121, 4903, 313, 9617, 3348, 3947, 2852, 2673, 5812, 6524, 1737, 2606, 9608, 634, 7867, 1554, 1413, 7926, 4041, 9154, 4258, 1734, 8296, 8140, 2252, 1259, 2695, 4161, 1139, 9930, 9876, 4599, 4293, 5634, 4563, 3277, 3237, 647, 3874, 9465, 7485, 2193, 4120, 3843, 9329, 8860, 263, 4146, 5717, 7128, 2173, 8749, 4001, 4371, 1399, 1293, 9546, 8929, 2301, 2144, 3564, 4784, 4273, 9802, 4010, 6222, 4487, 4514, 5770, 2532, 5358, 5820, 9487, 6857, 9817, 4255, 9815, 2616, 6538, 2967, 7298, 8878, 5761, 7413, 192, 4175, 2464, 4878, 325, 4785, 7834, 6751, 9528, 5000, 7430, 9610, 2145, 5365, 2452, 2813, 1980, 4135, 2408, 3584, 7601, 4260, 8311, 754, 5892, 9322, 3226, 5254, 3930, 6097, 554, 662, 4859, 7492, 2964, 2921, 769, 1688, 6527, 5473, 5220, 5642, 8205, 2952, 8093, 8446, 1371, 9475, 1963, 6177, 5318, 5827, 8414, 3345, 2572, 5980, 1036, 8130, 7289, 9849, 979, 5618, 7724, 7839, 9509, 8930, 4927, 9969, 1807, 5200, 2256, 9830, 3489, 7897, 2820, 1865, 1138, 2417, 319, 4492, 684, 8399, 1121, 9749, 9962, 5652, 1019, 5978, 9947, 8662, 1479, 2229, 5064, 5236, 1507, 1309, 8969, 5680, 7324, 9678, 9393, 8954, 1854, 8272, 8476, 205, 6963, 8033, 8820, 3532, 4036, 9469, 2469, 2611, 235, 7271, 6779, 6266, 9046, 3280, 9076, 5328, 2688, 5364, 9038, 5767, 4535, 3084, 8676, 6185, 9556, 4157, 7400, 3665, 2094, 7887, 4925, 8048, 195, 1331, 6768, 5666, 6911, 1683, 6806, 6315, 9653, 6242, 3586, 7592, 7807, 4761, 7961, 913, 404, 3431, 1086, 4201, 8586, 4468, 485, 3989, 5201, 5687, 101, 5496, 8200, 1103, 4429, 7184, 3918, 7137, 7803, 4490, 6579, 9726, 790, 6723, 596, 606, 2320, 8692, 722, 7775, 8515, 9863, 5707, 1284, 4627, 6420, 3802, 4331, 7645, 2376, 3259, 1463, 3132, 6737, 721, 8309, 1025, 6948, 509, 8412, 9398, 8315, 1897, 4000, 5646, 2534, 4963, 540, 3755, 8822, 4287, 6537, 11, 4663, 8174, 4494, 6636, 4989, 4572, 6839, 4822, 4892, 561, 9958, 4093, 8744, 9776, 2769, 414, 416, 6637, 9892, 4370, 4833, 5144, 880, 907, 9055, 5808, 2538, 5204, 4422, 7448, 1035, 1950, 9417, 7760, 1939, 8551, 5184, 2187, 1097, 9686, 525, 7923, 364, 9182, 3013, 3252, 2203, 1537, 3455, 2209, 6981, 8032, 831, 2096, 3113, 2137, 9938, 2010, 5686, 2920, 4382, 9611, 9909, 1610, 9440, 2868, 6120, 1267, 4521, 2578, 9923, 7143, 3981, 5882, 6886, 31, 6326, 2437, 6995, 3946, 2603, 3234, 8481, 3457, 2835, 6518, 8882, 8782, 3459, 2806, 4993, 8652, 6202, 8543, 3945, 9636, 7556, 1873, 7444, 7014, 8442, 4782, 5780, 9519, 7161, 1730, 4956, 9083, 3033, 9314, 9196, 4329, 2505, 9856, 7712, 3882, 1611, 9905, 2552, 7617, 6762, 8865, 5406, 2123, 5114, 4944, 8637, 5119, 2995, 2189, 3744, 7640, 4503, 344, 5071, 8213, 3318, 9933, 7783, 4098, 7860, 7074, 5591, 4820, 4011, 912, 3016, 4979, 6046, 8271, 3009, 3984, 1100, 710, 8214, 4666, 3503, 1618, 4954, 5300, 7917, 1870, 7443, 3504, 1516, 4330, 7258, 3383, 265, 9353, 5155, 2527, 6028, 8756, 9260, 2953, 7387, 7423, 9130, 3787, 5550, 1880, 9199, 8891, 8604, 4687, 2444, 8828, 199, 8588, 818, 4032, 3477, 7462, 1564, 215, 6833, 9723, 5512, 2792, 2872, 6143, 4734, 8031, 8705, 8888, 7111, 6682, 1559, 8787, 1434, 2660, 1937, 7563, 6939, 4527, 3527, 9713, 8622, 1178, 342, 9159, 3055, 6021, 7198, 4728, 9463, 333, 2938, 805, 6058, 4406, 908, 9944, 4259, 6540, 7344, 3698, 2282, 2966, 1458, 5606, 118, 8413, 5625, 185, 1460, 300, 2217, 4741, 6852, 2923, 6894, 8616, 8050, 1362, 3650, 7232, 7797, 6124, 697, 7445, 5251, 6239, 2432, 9360, 5936, 1266, 7802, 8388, 2438, 3949, 5121, 5425, 3895, 2776, 7904, 4990, 6287, 846, 4144, 6333, 3262, 5579, 7105, 4203, 7032, 8221, 8631, 8644, 2368, 5649, 3845, 7318, 9750, 126, 1274, 7864, 5417, 6681, 5349, 980, 958, 6815, 2706 ]
B: 90000

Getting failed for the above test case.

**Review by venkatyuva — CHANGES_REQUESTED** [review]
We shall discuss on this if you have more doubts on this review comments.

File: src/main/java/org/example/list/SinglyLinkedList.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        public Node(int val) {
+            this.val = val;
+            this.next = null;
+        }
+    }
+
+    public Node create(int val) {
+        Node node = new Node(val);
+
+        Node tailNode = this.getTailNode();
```
[the remark this answers is not in the document]
Comment (madhusdhnn) [reply src/main/java/org/example/list/SinglyLinkedList.java]: fixed

File: src/main/java/org/example/list/SinglyLinkedList.java [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    public void deleteNext(Node node) {
+        node.setNext(null);
+        this.decrementSize();
+    }
+
+    public void reset() {
+        this.head = null;
+    }
+
+    private void decrementSize() {
```
[the remark this answers is not in the document]
Comment (madhusdhnn) [reply src/main/java/org/example/list/SinglyLinkedList.java:88]: removed

File: src/main/java/org/example/list/SinglyLinkedList.java [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+        return str.append("]").toString();
+    }
+
+    private Node getTailNode() {
+        if (this.head == null) {
+            return null;
+        }
+
+        Node temp = this.head;
+        while (temp.next != null) {
```
[the remark this answers is not in the document]
Comment (madhusdhnn) [reply src/main/java/org/example/list/SinglyLinkedList.java]: fixed

File: src/main/java/org/example/problems/ArrayOps.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
+package org.example.problems;
+
+public class ArrayOps {
+
+    public int maxSumNonAdjacent(int[] arr) {
```
[the remark this answers is not in the document]
Comment (madhusdhnn) [reply src/main/java/org/example/problems/ArrayOps.java:5]: yes. If we include, what is the best sum we can do. or if we exclude what is the best we can do. Max of them is the result

File: src/main/java/org/example/problems/BookList.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+import lombok.ToString;
+import org.example.list.SinglyLinkedList;
+
+@ToString
+public class BookList {
+
+    private final SinglyLinkedList linkedList = new SinglyLinkedList();
+
+    public void createBook(int pages) {
+        this.linkedList.create(pages);
```
[the remark this answers is not in the document]
Comment (madhusdhnn) [reply src/main/java/org/example/problems/BookList.java:14]: fixed

File: src/main/java/org/example/problems/PizzaEating.java
Comment (madhusdhnn) [reply src/main/java/org/example/problems/PizzaEating.java:1]: Fixed

Comment (venkatyuva) [reply src/main/java/org/example/problems/PizzaEating.java:1]: Are you getting 114 as output? I am still getting 93 as output with your code.

Comment (madhusdhnn) [reply src/main/java/org/example/problems/PizzaEating.java:1]: yes 114

**Review by venkatyuva — APPROVED** [review]

**Pull request merged**